src/tests/suite/daos_test_common.c

```diff
@@ -1235,3 +1235,44 @@ int wait_and_verify_blobstore_state(uuid_t bs_uuid, char *expected_state,
 
 	return -DER_TIMEDOUT;
 }
+
+#define MAX_POOL_TGT_STATE_WAIT	   5 /* 5sec sleep between tgt state queries */
+#define MAX_POOL_TGT_STATE_RETRY   24 /* max timeout of 24 * 5sec= 2min */
+
+int wait_and_verify_pool_tgt_state(daos_handle_t poh, int tgtidx, int rank,
+				   char *expected_state)
+{
+	daos_target_info_t	tgt_info = { 0 };
+	int			retry_cnt;
+	int			rc;
+
+	retry_cnt = 0;
+	while (retry_cnt <= MAX_POOL_TGT_STATE_RETRY) {
+		char *expected_state_dup = strdup(expected_state);
+		char *state = strtok(expected_state_dup, "|");
+
+		rc = daos_pool_query_target(poh, tgtidx, rank, &tgt_info, NULL);
+			if (rc)
+				return rc;
+
+		/* multiple states not present in expected_state str */
```

Comment:
I'm wondering why don't you pass in integer state (instead of char *), so will be much simpler for state checking here.